Source/WTF/wtf/MediaTime.h

```diff
@@ -75,21 +75,11 @@ class WTF_EXPORT_PRIVATE MediaTime final {
     MediaTime operator-(const MediaTime& rhs) const;
     MediaTime operator-() const;
     MediaTime operator*(int32_t) const;
-    bool operator<(const MediaTime& rhs) const { return compare(rhs) == LessThan; }
-    bool operator>(const MediaTime& rhs) const { return compare(rhs) == GreaterThan; }
-    bool operator==(const MediaTime& rhs) const { return compare(rhs) == EqualTo; }
-    bool operator>=(const MediaTime& rhs) const { return compare(rhs) >= EqualTo; }
-    bool operator<=(const MediaTime& rhs) const { return compare(rhs) <= EqualTo; }
     bool operator!() const;
     explicit operator bool() const;
 
-    typedef enum {
-        LessThan = -1,
-        EqualTo = 0,
-        GreaterThan = 1,
-    } ComparisonFlags;
-
-    ComparisonFlags compare(const MediaTime& rhs) const;
+    WTF_EXPORT_PRIVATE friend std::weak_ordering operator<=>(const MediaTime&, const MediaTime&);
+    friend bool operator==(const MediaTime& a, const MediaTime& b) { return is_eq(a <=> b); }
```

Comment:
What's the benefit of using the friend version of these rather than the member version?

Reply:
If I remember correctly the benefits is when you do `a == b` it works even if a is not a MediaTime (but a type convertible to a MediaTime). I remember Darin recommended we prefer this friend version when we started defaulting operator==() in a lot of places so I'm following the same pattern for the spaceship operator.